### DISCUSSION TITLE
HID-Instance change after disconnecting device

### DISCUSSION THREAD

**Discussion by vrs99-hub** [nefarius/HidHide#50] [Q&A] [answered]
I just wanted to make sure about the following. I'm trying to use the HidHideCLI to automatically hide (some) controllers before starting certain games. That means settings up scripts with the HID-Instances of those controllers.

Unfortunately, I noticed that these HID-Instances change after disconnecting and reconnecting a controller. Does that mean that there is no reliable way to have a CLI script that will always work for the same physical controllers regardless of when it has been connected/disconnected to the computer?

Thanks for any info on this.

**Comment by nefarius** [comment, marked as answer]
The Instance ID is auto-generated by Windows and depends on a few factors, mainly how the device is connected. USB should be the most reliable, as long as you use the same port. Wireless gets a bit more complicated. So in essence, it depends I'm afraid.

Cheers
